src/decoder/mod.rs

```diff
@@ -53,23 +53,67 @@ impl OutputInfo {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct Limits {
+    /// max number of pixels: `width * height` (default: 67M = 2<sup>26</sup>)
+    pixels: u64,
+}
+
+impl Default for Limits {
+    fn default() -> Limits {
+        Limits {
+            pixels: 1 << 26,
+        }
+    }
+}
+
+impl Limits {
+    pub fn set_pixels(&mut self, pixels: u64) {
+        self.pixels = pixels;
+    }
+}
+
 /// PNG Decoder
 pub struct Decoder<R: Read> {
     /// Reader
     r: R,
     /// Output transformations
     transform: Transformations,
+    /// Images that are considered too big
+    limits: Limits,
 }
 
 impl<R: Read> Decoder<R> {
     pub fn new(r: R) -> Decoder<R> {
         Decoder {
             r: r,
             transform: ::Transformations::EXPAND | ::Transformations::SCALE_16 | ::Transformations::STRIP_16,
-
+            limits: Limits::default(),
         }
     }
 
+    /// Images that are considered too big
+    ///
+    /// ```
+    /// use std::fs::File;
+    /// use png::{Decoder, Limits};
+    /// // This image is 32x32 pixels, so it's more than four pixels in size.
+    /// let mut limits = Limits::default();
+    /// limits.set_pixels(4);
+    /// let mut decoder = Decoder::new(File::open("tests/pngsuite/basi0g01.png").unwrap());
+    /// decoder.set_limits(limits);
+    /// assert!(decoder.read_info().is_err());
+    /// // This image is 32x32 pixels, so it's exactly 1024 pixels in size.
+    /// let mut limits = Limits::default();
+    /// limits.set_pixels(1024);
+    /// let mut decoder = Decoder::new(File::open("tests/pngsuite/basi0g01.png").unwrap());
+    /// decoder.set_limits(limits);
+    /// assert!(decoder.read_info().is_ok());
+    /// ```
+    pub fn set_limits(&mut self, limits: Limits) {
+        self.limits = limits;
```

Comment:
How about using `new_with_limits(r: R, limits: Limits)` method instead? It will allow to create decoder in one line and `new` can be defined as `Self::new_with_limits(r, Default::default())`.